snippets/base/management/commands/generate_bundles.py

```diff
@@ -45,63 +46,70 @@ def handle(self, *args, **options):
-
-                if settings.BUNDLE_BROTLI_COMPRESS:
-                    content_file = ContentFile(brotli.compress(bundle_content))
-                    content_file.content_encoding = 'br'
-                else:
-                    content_file = ContentFile(bundle_content)
-
-                default_storage.save(filename, content_file)
-                self.stdout.write(self.style.SUCCESS('Writing bundle {}'.format(filename)))
+        distribution_bundles_to_process = DistributionBundle.objects.filter(
+            distributions__jobs__in=total_jobs
+        ).distinct()
+
+        for distribution_bundle in distribution_bundles_to_process:
+            distributions = distribution_bundle.distributions.all()
+
+            for channel, locale in combinations_to_process:
+                channel_jobs = Job.objects.filter(
+                    status=Job.PUBLISHED).filter(**{
+                        'targets__on_{}'.format(channel): True,
+                        'distribution__in': distributions,
+                    })
+
+                locales_to_process = [
+                    key.lower() for key in product_details.languages.keys()
+                    if key.lower().startswith(locale)
+                ]
+
+                for locale_to_process in locales_to_process:
+                    filename = 'Firefox/{channel}/{locale}/{distribution}.json'.format(
+                        channel=channel,
+                        locale=locale_to_process,
+                        distribution=distribution_bundle.code_name,
+                    )
+                    filename = os.path.join(settings.MEDIA_BUNDLES_PREGEN_ROOT, filename)
+                    full_locale = ',{},'.format(locale_to_process.lower())
+                    splitted_locale = ',{},'.format(locale_to_process.lower().split('-', 1)[0])
+                    bundle_jobs = channel_jobs.filter(
+                        Q(snippet__locale__code__contains=splitted_locale) |
+                        Q(snippet__locale__code__contains=full_locale))
+
+                    # If there 're no Published Jobs for the channel / locale /
+                    # distribution combination, delete the current bundle file if
+                    # it exists.
+                    if not bundle_jobs.exists():
+                        if default_storage.exists(filename):
+                            self.stdout.write('Removing {}'.format(filename))
+                            default_storage.delete(filename)
+                        continue
+
+                    data = [job.render() for job in bundle_jobs]
+                    bundle_content = json.dumps({
+                        'messages': data,
+                        'metadata': {
+                            'generated_at': datetime.utcnow().isoformat(),
+                            'number_of_snippets': len(data),
+                        }
+                    })
+
+                    # Convert str to bytes.
```

Comment:
Just an FYI. You may know this, but just in case you don't, [Django has some very useful functions for str/bytes stuff](https://docs.djangoproject.com/en/2.2/ref/utils/#module-django.utils.encoding) that just make it a little easier e.g. `force_bytes` and `force_string` as well as some counterparts that respect lazy objects.